NPE under some deployment configurations

Sometimes  a NPE occurs when loading the activator class:

Caused by: org.osgi.framework.BundleException: Error loading bundle activator.
	at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:760)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1005)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:357)
	at org.eclipse.osgi.container.Module.doStart(Module.java:589)
	at org.eclipse.osgi.container.Module.start(Module.java:457)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:471)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:117)
	... 26 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.loadBundleActivator(BundleContextImpl.java:799)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:752)
	... 32 more
Caused by: java.lang.NullPointerException
	at org.eclipse.jface.resource.JFaceResources.getResources(JFaceResources.java:209)
	at org.eclipse.jface.resource.ImageRegistry.<init>(ImageRegistry.java:148)
	at org.eclipse.jface.resource.ImageRegistry.<init>(ImageRegistry.java:120)
	at org.eclipse.swtchart.extensions.Activator.<init>(Activator.java:40)
	... 38 more
osgi>

--- a/org.eclipse.swtchart.extensions/src/org/eclipse/swtchart/extensions/Activator.java
+++ b/org.eclipse.swtchart.extensions/src/org/eclipse/swtchart/extensions/Activator.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2017, 2019 Lablicate GmbH.
+ * Copyright (c) 2017, 2020 Lablicate GmbH.
  *
  * This program and the accompanying materials are made
  * available under the terms of the Eclipse Public License 2.0
@@ -9,6 +9,7 @@
  * 
  * Contributors:
  * Dr. Philip Wenig - initial API and implementation
+ * Christoph Läubrich - lazy init of ImageRegistry
  *******************************************************************************/
 package org.eclipse.swtchart.extensions;
 
@@ -21,7 +22,10 @@ import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Path;
 import org.eclipse.jface.resource.ImageDescriptor;
 import org.eclipse.jface.resource.ImageRegistry;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.SWTException;
 import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.Display;
 import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleActivator;
 import org.osgi.framework.BundleContext;
@@ -36,13 +40,11 @@ public class Activator implements BundleActivator {
 	public static final String ICON_RESET = "ICON_RESET"; // $NON-NLS-1$ //$NON-NLS-1$
 	//
 	private static Activator plugin;
-	
-	private ImageRegistry imageRegistry = new ImageRegistry();
-
 	/**
 	 * The bundle associated this plug-in
 	 */
 	private Bundle bundle;
+	private ImageRegistry imageRegistry;
 
 	/**
 	 * The constructor
@@ -54,17 +56,18 @@ public class Activator implements BundleActivator {
 	 * (non-Javadoc)
 	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.BundleContext)
 	 */
+	@Override
 	public void start(BundleContext context) throws Exception {
 
 		plugin = this;
 		this.bundle = context.getBundle();
-		initializeImageRegistry();
 	}
 
 	/*
 	 * (non-Javadoc)
 	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#stop(org.osgi.framework.BundleContext)
 	 */
+	@Override
 	public void stop(BundleContext context) throws Exception {
 
 		plugin = null;
@@ -72,6 +75,9 @@ public class Activator implements BundleActivator {
 
 	public Image getImage(String key) {
 
+		if(imageRegistry == null) {
+			initializeImageRegistry();
+		}
 		return imageRegistry.get(key);
 	}
 
@@ -97,6 +103,11 @@ public class Activator implements BundleActivator {
 
 	private void initializeImageRegistry() {
 
+		if(Display.getCurrent() == null) {
+			throw new SWTException(SWT.ERROR_THREAD_INVALID_ACCESS);
+		} else {
+			imageRegistry = new ImageRegistry();
+		}
 		Map<String, String> imageHashMap = new HashMap<String, String>();
 		imageHashMap.put(ICON_SET_RANGE, "icons/16x16/set_range.gif"); // $NON-NLS-1$ //$NON-NLS-1$
 		imageHashMap.put(ICON_HIDE, "icons/16x16/hide.gif"); // $NON-NLS-1$ //$NON-NLS-1$

